kimchi/src/circuits/gate.rs

```diff
@@ -372,7 +380,7 @@ impl<F: PrimeField + SquareRootField> CircuitGate<F> {
             [F::zero(); COLUMNS]
         };
 
-        Ok(ArgumentWitness::<F> {
+        Ok(ArgumentWitness::<F, COLUMNS> {
```

Comment:
Nit: you can omit the generics here, the return type is already explicitly specified.